opal/corelib/string.rb

```diff
@@ -1575,7 +1575,7 @@ def upto(stop, excl = false, &block)
     return enum_for :upto, stop, excl unless block_given?
     stop = Opal.coerce_to(stop, String, :to_str)
     %x{
-      var a, b, s = self.toString();
+      var a, b, s = self.toString(), value;
```

Comment:
Here I realized that there's probably a bunch of methods that currently do not respect/support `break` values. This fixes `String#upto`, so that:
```ruby
'a'.upto('z'){break 'hello'} #=> "hello"
```